lib/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -364,6 +364,13 @@ void HdVP2Mesh::Sync(
     HdDirtyBits* dirtyBits,
     const TfToken& reprToken)
 {
+    // We don't create a repr for the selection token because this token serves
```

Comment:
There is an optimization in Hydra that would skip sync if a Rprim is invisible and has no visibility dirty bit:

https://github.com/PixarAnimationStudios/USD/blob/4b1162957b2ad219e1635c81de8343be490e41fc/pxr/imaging/lib/hd/rprim.cpp#L71

Due to this optimization, if we have a selection update before regular refresh, the scene dirty bit gets cleared in selection update and no sync can be invoked for the hiding proxy. Thus we need the earlier return for the selection update to reserver the scene dirty bits.